docs/decisions/0033-rag-in-sk.md

```diff
@@ -0,0 +1,297 @@
+3. Abstractions should not block developers from using vector DB of their choice with functionality, that cannot be achieved with provided interfaces or data types.
+
+## Out of scope
+
+Some of the RAG-related frameworks contain functionality to support full cycle of RAG pattern:
+
+1. **Read** data from specific resource (e.g. Wikipedia, OneDrive, local PDF file).
+2. **Split** data in multiple chunks using specific logic.
+3. **Generate** embeddings from data.
+4. **Store** data to preferred vector DB.
+5. **Search** data in preferred vector DB based on user query.
+6. **Ask** LLM a question based on provided data.
+
+As for now, Semantic Kernel has following experimental features:
+
+- `TextChunker` class to **split** data in chunks.
+- `ITextEmbeddingGenerationService` abstraction and implementations to **generate** embeddings using OpenAI and HuggingFace models.
+- Memory connectors to **store** and **search** data.
+
+Since these features are experimental, they may be deprecated in the future if the decisions for RAG pattern won't require to provide and maintain listed abstractions, classes and connectors in Semantic Kernel.
+
+Tools for data **reading** is out of scope as for now.
+
+## Considered Options
+
+### Option 1 [Supported] - Prompt concatenation
+
+This option allows to manually construct a prompt with data, so LLM can respond to query based on provided context. It can be achieved by using manual string concatenation or by using prompt template and Kernel arguments. Developers are responsible for integration with vector DB of their choice, data search and prompt construction to send it to LLM.
+
+This approach doesn't include any memory connectors in Semantic Kernel out-of-the-box, but at the same time it gives an opportunity for developers to handle their data in the way that works for them the best.
+
+String concatenation:
+
+```csharp
+var kernel = Kernel.CreateBuilder()
+    .AddOpenAIChatCompletion("model-id", "api-key")
+    .Build();
+
+var builder = new StringBuilder();
+
+// User is responsible for searching the data in a way of their choice, this is an example how it could look like.
+var data = await this._vectorDB.SearchAsync("Company budget by year");
+
+builder.AppendLine(data);
+builder.AppendLine("What is my budget for 2024?");
+
+var result = await kernel.InvokePromptAsync(builder.ToString());
+```
+
+Prompt template and Kernel arguments:
+
+```csharp
+var kernel = Kernel.CreateBuilder()
+    .AddOpenAIChatCompletion("model-id", "api-key")
+    .Build();
+
+// User is responsible for searching the data in a way of their choice, this is an example how it could look like.
+var data = await this._vectorDB.SearchAsync("Company budget by year");
+
+var arguments = new KernelArguments { ["budgetByYear"] = data };
+
+var result = await kernel.InvokePromptAsync("{{budgetByYear}} What is my budget for 2024?", arguments);
+```
+
+### Option 2 [Supported] - Memory as Plugin
```

Comment:
My preference would be to support Option 2 as the main/preferred route since it strengthens the plugin story and also leverages prompt templates the primary asset (which would allow RAG templates to be shared between Python/.NET/Java)

Reply:
That makes sense!